core/network/firewall/rule.go

```diff
@@ -221,3 +221,33 @@ nextRule:
 
 	return uniqueRules
 }
+
+// RemoveCIDRsMatchingAddressType returns a new list of rules where any CIDR
+// whose address type corresponds to the specified AddressType argument has
+// been removed.
+func (rules IngressRules) RemoveCIDRsMatchingAddressType(removeAddrType network.AddressType) IngressRules {
```

Comment:
Passing the address type we want to remove as a predicate allows us to do more things as opposed to the above proposal. For example:
- Split rules into an IPV4 and IPV6 list (`v4list, v6List := RemoveCIDRsMatchingAddressType(network.IPV6Address), RemoveCIDRsMatchingAddressType(network.IPV4Address)`)
- Get a list of the IPV6 rules that we filtered out (`RemoveCIDRsMatchingAddressType(network.IPV4Address)`)

Since this is part of core/network I wanted it to be a bit more generic similar to how other types in the same package operate.